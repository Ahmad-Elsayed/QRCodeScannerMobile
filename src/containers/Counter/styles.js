import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  containerStyle: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-evenly',
  },
  countTextStyle: {
    color: '#C71585',
    fontSize: 20,
    fontWeight: 'bold',
  },
  btnContainerStyle: {
    width: '100%',
    justifyContent: 'space-around',
    flexDirection: 'row',
  },
  btnStyle: {
    paddingHorizontal: 30,
  },
});
