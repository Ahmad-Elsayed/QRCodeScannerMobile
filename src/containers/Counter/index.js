import React, { useState } from 'react';
import { View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { Button, Text } from 'components';
import { mainRoutes } from 'src/constants/routeNames';
import styles from './styles';
import { useSelector } from 'react-redux';

const QRScanner = () => {
  const { navigate } = useNavigation();
  const { count } = useSelector(state => state.counter);
  return (
    <View style={styles.containerStyle}>
      <Text style={styles.countTextStyle}>{count}</Text>
      <View style={styles.btnContainerStyle}>
        <Button
          title="Back"
          containerStyle={styles.btnStyle}
          onPress={() => {
            navigate(mainRoutes.home);
          }}
        />
        <Button
          title="Scan"
          containerStyle={styles.btnStyle}
          onPress={() => navigate(mainRoutes.qrCodeScanner)}
        />
      </View>
    </View>
  );
};

export default QRScanner;
