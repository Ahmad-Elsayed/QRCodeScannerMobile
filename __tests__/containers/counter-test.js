/**
 * @format
 */
import 'react-native';

import React from 'react';

import renderer from 'react-test-renderer';
import Counter from 'containers/Counter';

it('renders correctly', () => {
  const counter = renderer.create(<Counter />).toJSON();
  expect(counter).toMatchSnapshot();
});
