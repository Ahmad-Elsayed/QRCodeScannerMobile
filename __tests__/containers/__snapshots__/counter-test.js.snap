// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-evenly",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
          "textAlign": "left",
        },
        Object {
          "color": "#C71585",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#45b1e8",
            "borderRadius": 6,
            "paddingHorizontal": 30,
            "paddingVertical": 12,
          },
          false,
          Object {
            "paddingHorizontal": 30,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "textAlign": "left",
            },
            Object {
              "color": "#fff",
              "fontSize": 13,
              "fontWeight": "500",
            },
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#45b1e8",
            "borderRadius": 6,
            "paddingHorizontal": 30,
            "paddingVertical": 12,
          },
          false,
          Object {
            "paddingHorizontal": 30,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 15,
              "textAlign": "left",
            },
            Object {
              "color": "#fff",
              "fontSize": 13,
              "fontWeight": "500",
            },
          ]
        }
      >
        Scan
      </Text>
    </View>
  </View>
</View>
`;
